scipio/src/executor.rs

```diff
@@ -247,6 +247,7 @@ struct ExecutorQueues {
     executor_index: usize,
     last_vruntime: u64,
     preempt_timer_duration: Duration,
+    spin_before_park: Option<Duration>,
```

Comment:
good first step. But how is the user supposed to select this?

Somehow the user of the library must choose that. I suspect that the best way is instead of passing
just a cpu binding option to the executor, we will pass a struct with options.

For now it would contain the pinning settings plus the spin setting.

With time, if we add more, the API would stay the same and the creation of LocalExecutor wouldn't break.

Two important things are also missing from this PR:
- A test. We can use linux's `getrusage(2)` to make sure this has a real effect.
- Storing a metric in the executor that captures real usage vs total usage

Reply:
done